internal/services/queueService.go

```diff
@@ -56,31 +73,173 @@ func (qs *queueService) Listen() {
+				}
+				continue
+			}
+
+			err = qs.workerPool.ProcessTask(msg.ReplyTo, queueMessage.MessageID, task)
+			if err != nil {
+				qs.logger.Errorf("Failed to process task message: %s", err)
+				if err == errors.ErrFailedToGetFreeWorker {
+					err = qs.RequeueTaskWithHigherPriority(queueMessage)
+					if err != nil {
+						qs.logger.Errorf("Failed to requeue task with higher priority: %s", err)
+					}
+				} else {
+					err = PublishErrorToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, err)
+					if err != nil {
+						qs.logger.Errorf("Failed to publish error message: %s", err)
+					}
+				}
+				continue
+			}
+
+		} else if queueMessage.Type == constants.QueueMessageTypeStatus {
+			qs.logger.Infof("Processing status message")
+			status := qs.workerPool.GetWorkersStatus()
+			payload, err := json.Marshal(status)
+			if err != nil {
+				qs.logger.Errorf("Failed to marshal status message: %s", err)
+				err = PublishErrorToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, err)
+				if err != nil {
+					qs.logger.Errorf("Failed to publish error message: %s", err)
+				}
+				continue
+			}
+
+			err = PublishSucessToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, payload)
+			if err != nil {
+				qs.logger.Errorf("Failed to publish status message: %s", err)
+				err = PublishErrorToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, err)
+				if err != nil {
+					qs.logger.Errorf("Failed to publish error message: %s", err)
+				}
+			}
+
+		} else if queueMessage.Type == constants.QueueMessageTypeHandshake {
+			qs.logger.Infof("Processing handshake message")
+
+			supportedLanguages := languages.GetSupportedLanguagesWithVersions()
+			payload, err := json.Marshal(supportedLanguages)
+			if err != nil {
+				qs.logger.Errorf("Failed to marshal supported languages: %s", err)
+				err = PublishErrorToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, err)
+				if err != nil {
+					qs.logger.Errorf("Failed to publish error message: %s", err)
+				}
+				continue
+			}
+
+			err = PublishSucessToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, payload)
+			if err != nil {
+				qs.logger.Errorf("Failed to publish supported languages: %s", err)
+				err = PublishErrorToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, err)
+				if err != nil {
+					qs.logger.Errorf("Failed to publish error message: %s", err)
+				}
+			}
 		}
+	}
 
-		responseJSON, err := json.Marshal(response)
-		if err != nil {
-			qs.logger.Errorf("Failed to marshal response: %s", err)
-			continue
-		}
+}
 
-		qs.logger.Infof("Sending response: %s", response)
+func PublishErrorToResponseQueue(channel *amqp.Channel, responseQueueName, messageType, messageID string, err error) error {
```

Comment:
This is not properly documented. In README there is different schema. There should be indicator if this is error message, for example, message type of field _ok_